### PR TITLE
Add lib to sign and publish artifacts to artifacts.opensearch.org

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
This change adds the library to sign and publish the artifacts to artifacts.opensearch.org. The role management is manual as of today but we are looking into automating the same.

### Issues Resolved
required for https://github.com/opensearch-project/opensearch-build/issues/3112

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: vars/publishToArtifactsProdBucket.groovy [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * compatible open source license.
+ */
+
+ /**
+ * Library to sign and upload artifacts to artifacts.opensearch.org
+@param Map[assumedRoleName] <required> - IAM role to be assumed for uploading artifacts
+@param Map[source] <required> - Path to yml or artifact file.
+@param Map[destination] <required> - Artifact type in the manifest, [type] is required for signing yml.
+@param Map[signingPlatform] <optional> - The distribution platform for signing. Defaults to linux
+@param Map[sigtype] <optional> - signature type. Defaults to '.sig'
```
Comment (rishabh6788) [inline vars/publishToArtifactsProdBucket.groovy:16]: Nit: `sigType`

File: vars/publishToArtifactsProdBucket.groovy [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+@param Map[overwrite]<optional> - Allow output artifacts to overwrite the existing artifacts. Defaults to false
```
Comment (rishabh6788) [inline vars/publishToArtifactsProdBucket.groovy:17]: Nit: `overWrite`

**Review by rishabh6788 — APPROVED** [review]

File: vars/publishToArtifactsProdBucket.groovy (hunk 2 above)
Comment (gaiksaya) [reply vars/publishToArtifactsProdBucket.groovy:17]: Hey! So these args are supplied as it is to `sign.sh` so has to be `sigtype` and `overwrite`: https://github.com/opensearch-project/opensearch-build/blob/main/src/sign_workflow/sign_args.py#L45-L57

Comment (gaiksaya) [reply vars/publishToArtifactsProdBucket.groovy:17]: Trying to keep the args as same as possible. Wondering if I should change `signingPlatform` to just `platform`?

Comment (gaiksaya) [reply vars/publishToArtifactsProdBucket.groovy]: Fixed!

**Pull request merged**